src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -98,6 +98,21 @@ _mongoc_client_pool_set_internal_tls_opts (
 
 mongoc_client_pool_t *
 mongoc_client_pool_new (const mongoc_uri_t *uri)
+{
+   mongoc_client_pool_t *pool;
+   bson_error_t error = {0};
+
+   if (!(pool = mongoc_client_pool_new_with_error (uri, &error)) &&
+       0 != strcmp (error.message, "")) {
+      MONGOC_ERROR ("%s", error.message);
+   }
```

Comment:
Per my comment on `mongoc_client_pool_new.rst`, I may need to change this such that `mongoc_client_pool_new` _always` returns a non-NULL pointer to avoid a BC break. Either way, I think it'd be sensible to keep the behavior that logs an error if we have something to report.

Reply:
I think this looks good as written.